src/App.ts

```diff
@@ -6,84 +6,158 @@
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+  redisClient: redis.RedisClient,
+  pagoPAClient: pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient
 ): void {
+  const jsonParser = bodyParser.json();
   app.get(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_CHECK,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
     (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Check Request (GET)...");
       toExpressHandler(
-        PaymentController.checkPaymentToPagoPa(
-          config.PAGOPA,
-          verificaRPTPagoPaClient
-        )
+        PaymentController.getPaymentInfo(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
     (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Activation Request (POST)...");
       toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
+        PaymentController.activatePayment(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    (req: express.Request, res: express.Response) => {
+      toExpressHandler(PaymentController.getActivationStatus(redisClient))(
+        req,
+        res
+      );
+    }
+  );
+
+  // Endpoint for OpenAPI handler
+  app.get("/specs/api/v1/swagger.json", GetOpenapi(publicApiV1Specs));
+}
+
+/**
+ * Define and return a method to start a SOAP Server for PagoPA
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {redisTimeout} redisTimeout - The timeout to use for information stored into redis
+ * @return {(app: core.Express) => soap.Server} A method to execute for start server listening
+ */
+function getSoapServer(
+  redisClient: redis.RedisClient,
+  redisTimeout: number
+): (app: core.Express) => soap.Server {
+  // Configuration for SOAP endpoints and callback handler
+  const service = {
+    FespCdService: {
+      FespCdPortType: {
+        cdInfoPagamento: async (
+          iCdInfoPagamentoInput: IcdInfoPagamentoInput,
+          callback: (iCdInfoPagamentoOutput: IcdInfoPagamentoOutput) => void
+        ): Promise<IcdInfoPagamentoOutput> => {
+          const iCdInfoPagamentoOutput = await PaymentController.setActivationStatus(
+            iCdInfoPagamentoInput,
+            redisTimeout,
+            redisClient
+          );
+          callback(iCdInfoPagamentoOutput);
+          return iCdInfoPagamentoOutput;
+        }
+      }
+    }
+  };
+  // Retrieve the wsdl related to the endpoint and define a starter function to return
+  const wsdl = fs.readFileSync(`${FespCdService_WSDL_PATH}`, "UTF-8");
+  return (app: core.Express): soap.Server => {
```

Comment:
wait, this is what [this function](https://github.com/teamdigitale/italia-pagopa-api/blob/master/lib/servers.ts#L23) is for - did you read/understand the code in `italia-pagopa-api` for building a server?

Reply:
attachFespCdServer was not available on api.

I just implemented it on
https://github.com/teamdigitale/italia-pagopa-api/tree/Manual-fix-for-italia-pagopa-proxy